inception-workload/src/main/java/de/tudarmstadt/ukp/inception/workload/model/WorkloadManagementService.java

```diff
@@ -17,12 +17,32 @@
  */
 package de.tudarmstadt.ukp.inception.workload.model;
 
+import java.util.List;
+
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationDocument;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationDocumentState;
 import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.User;
 
+/**
+ * Interface for all required DB calls. Short explanation given in the
+ * Interface implementation class WorkloadManagementServiceImpl
+ */
 public interface WorkloadManagementService
 {
     WorkloadManager getOrCreateWorkloadManagerConfiguration(Project aProject);
 
-    void setWorkloadManagerConfiguration(String workloadType, Project aProject);
-    void setTraits(String aTraits, Project aProject);
+    void setWorkloadManagerConfiguration(String aWorkloadType, Project aProject);
```

Comment:
Normally, we don't have methods for setting individual DB columns in the services. We work on the entities and then persist the entities.

So in this case, we would use `getOrCreateWorkloadManagerConfiguration()`  to obtain a `WorkloadManager` entity, then we would set the workload type and/or traits on that entity, and then we would eventually save the entity.

Typically, we load the entity into a form. Then the user makes modifications on the form. And when the user presses save, we persist the entity that is model of the form.

So in the service beans, we usually have the typically only have methods for loading, listing and saving entities but none that manipulate entities.

Could you please change this so that setWorkloadMangagerConfiguration goes away and we work only with the WorkloadManager entity and methods for loading and saving it?

Reply:
Ahh ok, thanks for the information and description. Sure no problem. I changed it now to your likings.